src/Adapter/Order/CommandHandler/UpdateProductInOrderHandler.php

```diff
@@ -145,6 +146,12 @@ private function assertProductCanBeUpdated(
         Order $order,
         OrderInvoice $orderInvoice = null
     ) {
+        // assert product exists
+        $product = new Product($orderDetail->product_id);
+        if ($product->id !== (int) $orderDetail->product_id) {
+            throw new CannotUpdateProductInOrderException('You cannot edit the price of a product that no longer exists in your catalog.');
```

Comment:
The exception name seems too generic as there are many reasons why the product can not be updated (see all the rest of the function). This exception should reflect this particular reason, something like `ProductInOrderNotFoundException` `CannotFindProductInOrderException` or similar